lib/puppet/type/file.rb

```diff
@@ -719,7 +716,9 @@ def remove_existing(should)
   end
 
   def retrieve
-    if source = parameter(:source)
+    # `checksum_value` implies explicit management of all metadata, so skip metadata
+    # retrieval. Otherwise, if source is set, retrieve metadata for source.
+    if (source = parameter(:source)) && property(:checksum_value).nil?
```

Comment:
Would it ever make sense for someone to specify `source` and `checksum_value` (so it would have to be a file, and not a recursive directory), but rely on the `source` to provide permissions? I'm thinking not, given puppet's default behavior of ignoring source permissions. If you specify `checksum_value`, I think it makes sense to also specify permissions that you care about. Maybe add a note to `checksum_value` description?